### ISSUE TITLE
Document how to instrument SQLAlchemy manually for PostgreSQL

### ISSUE THREAD

**Issue by AkselAllas** [open-telemetry/opentelemetry-python-contrib#1648]
How can I instrument PostgreSQL SQLAlchemy manually?
I got it working using autoinstrumentation, but not using manual instrumentation. (Which I need since some GCP trace propagation doesn't work otherwise)

I tried these (but doesn't work, only connect span is created):
```
from opentelemetry.instrumentation.sqlalchemy import SQLAlchemyInstrumentor
SQLAlchemyInstrumentor().instrument()
```

```
dbapi.trace_integration(engine, "connect", "PostgreSQL")
```

**Comment by jmsierra** [comment]
+1
I think the same case it's happening to me. I've got a custom application (not based in Flask or any other framework) and so I need Manual Instrumentation. As part of this application I'm using SQLAlchemy and whereas Manual Instrumentation is working pretty nicely for the application, the SQLAlchemy part doesn't seem to work or, at least, not sending the traces to opentelemetry collector as the Manual does.

```
self.engine = create_engine(cstr, convert_unicode=False, echo=False, max_identifier_length=30)
SQLAlchemyInstrumentor().instrument(engine=self.engine)
```

**Comment by AkselAllas** [comment]
🤦 
The following works:
```
SQLAlchemyInstrumentor().instrument(engine=self.engine)
```

**Comment by vanchaxy** [comment]
If you encounter this issue while using AsyncEngine, please be aware that you need to use `SQLAlchemyInstrumentor().instrument(engine=engine.sync_engine)` for instrumentation. This is because AsyncEngine does not support event listeners.

**Comment by ocelotl** [comment]
I asked Claude to review this issue and Claude suggests it be closed with this explanation:

> Docstring now documents `instrument(engine=...)` incl. `engine.sync_engine` for async.
>
> Proof in `main`: [`sqlalchemy/__init__.py#L32-L39`](https://github.com/open-telemetry/opentelemetry-python-contrib/blob/94e851dfe333d47fdb52edb82635330668af284e/instrumentation/opentelemetry-instrumentation-sqlalchemy/src/opentelemetry/instrumentation/sqlalchemy/__init__.py#L32-L39)

To guard against a mistaken suggestion, Claude then independently re-verified this conclusion:

> Verified: `sqlalchemy/__init__.py` documents `instrument(engine=...)` including `engine.sync_engine` for async engines.

Claude evaluated this issue with high confidence that it can be closed.

Claude has classified this issue as already resolved (the requested fix or feature has already shipped).

The intention of this comment is to make it easier for a human being to review this issue before making a decision.

**Comment by tammy-baylis-swi** [comment]
Agreed, this is an old issue and the docs have been updated since original reporting.

**Comment by nioncode** [comment]
I think the issue still applies. The docs demonstrate that you *can* pass an engine, but they don't describe why you might want to pass the engine or that in fact you most likely want to pass the engine to get any meaningful results.
